From dd642c74ddc5aaacf42ef94977e11e6d5948a363 Mon Sep 17 00:00:00 2001
From: Dmitry Osipenko <digetx@gmail.com>
Date: Thu, 20 Oct 2016 18:47:05 +0300
Subject: [PATCH 2/5] tegra: Fix drm_tegra_job_free() mem corruption

The drm_tegra_pushbuf_free() removes pushbuf element from the list, so the
DRMLISTFOREACHENTRYSAFE() should be used. That was spotted by valgrind.

Signed-off-by: Dmitry Osipenko <digetx@gmail.com>
---
 tegra/job.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tegra/job.c b/tegra/job.c
index 23e34c3f..d26d35dd 100644
--- a/tegra/job.c
+++ b/tegra/job.c
@@ -73,11 +73,12 @@ int drm_tegra_job_new(struct drm_tegra_job **jobp,
 int drm_tegra_job_free(struct drm_tegra_job *job)
 {
 	struct drm_tegra_pushbuf_private *pushbuf;
+	struct drm_tegra_pushbuf_private *temp;
 
 	if (!job)
 		return -EINVAL;
 
-	DRMLISTFOREACHENTRY(pushbuf, &job->pushbufs, list)
+	DRMLISTFOREACHENTRYSAFE(pushbuf, temp, &job->pushbufs, list)
 		drm_tegra_pushbuf_free(&pushbuf->base);
 
 	free(job->relocs);
-- 
2.12.0

